### PR TITLE
🌱 Refactor template addon get values func

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #https://github.com/open-cluster-management-io/ocm/pull/417#discussion_r1574335522

### CODE REVIEW

File: pkg/addon/templateagent/values_test.go [hunk 1]
```diff
@@ -225,21 +268,21 @@ func TestGetValues(t *testing.T) {
...
 
-			defaultValue, overrideValue, builtInValue, err := agentAddon.getValues(cluster, addon, addonTemplate)
+			presetValues, overrideValues, privateValues, err := agentAddon.getValues(cluster, addon, addonTemplate)
 			if err != nil {
 				t.Fatal(err)
 			}
-			fmt.Printf("defaultValue is %v\n", defaultValue)
-			if !orderedValueEquals(defaultValue, c.expectedDefault) {
-				t.Errorf("default value is not correct, expect %v, got %v", c.expectedDefault, defaultValue)
+			fmt.Printf("presetValues is %v\n", presetValues)
```
Comment (qiujian16) [inline pkg/addon/templateagent/values_test.go:275]: this line should be removed actually. It is for debug...

**Comment by zhujian7** [comment]
@qiujian16 PTAL

**Pull request merged**